File: 4107-find-missing-elements/find-missing-elements.java
class Solution {
    public List<Integer> findMissingElements(int[] nums) {
        List <Integer> list = new ArrayList<>();
        int min = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;
        HashSet <Integer> set = new HashSet <>();
        for (int i : nums){
            min = Math.min(i, min);
            max = Math.max(i, max);
            set.add(i);
        }
        for (int i = min+1; i < max; i++ ){
            if(!set.contains(i)){
                list.add(i);
            }
        }
        return list;
    }
}

File: 4107-find-missing-elements/find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        unordered_set <int> unique;
        vector <int> missing;
        int max = *max_element(nums.begin(), nums.end());
        int min = *min_element(nums.begin(), nums.end());
        for(int i: nums){
            unique.insert(i);
        }
        for(int i = min; i <= max; i++){
            if(!unique.count(i)) missing.push_back(i);
        }
        return missing;
    }
};